test/unit/muxed_account_test.js

```diff
@@ -0,0 +1,59 @@
+describe('muxed account abstraction works', function() {
+  const PUBKEY = 'GA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJVSGZ';
+  const MPUBKEY_ZERO =
+    'MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAACJUQ';
+  const MPUBKEY_ID =
+    'MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAABUTGI4';
+  const MPUBKEY_ID2 =
+    'MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAABQHF6OU';
+
+  const performBasicInvariantChecks = function(mux, id, expectedM) {
+    expect(mux.accountId()).to.equal(PUBKEY);
+    expect(mux.address()).to.equal(expectedM);
+    expect(mux.id()).to.equal(id);
+
+    const muxXdr = mux.asXDRObject();
+    expect(muxXdr.switch()).to.equal(
+      StellarBase.xdr.CryptoKeyType.keyTypeMuxedEd25519()
+    );
+
+    const innerMux = muxXdr.med25519();
+    expect(innerMux.ed25519()).to.eql(
+      StellarBase.StrKey.decodeEd25519PublicKey(PUBKEY)
+    );
+    expect(innerMux.id()).to.eql(StellarBase.xdr.Uint64.fromString(id));
+
+    return muxXdr;
+  };
+
+  it('can construct from a G address', function() {
+    const mux = new StellarBase.MuxedAccount(PUBKEY);
+    performBasicInvariantChecks(mux, '0', MPUBKEY_ZERO);
+  });
+  it('can construct from an M address', function() {
+    const mux = new StellarBase.MuxedAccount(MPUBKEY_ZERO);
```

Comment:
Should 'id' be required?

Reply:
It's hard to tell what the use-cases of these APIs will be just yet, but I envisioned that if somebody decided to create a bunch of muxed accounts from a single public key, they'd want some "master" muxed version that just has an ID of zero. There's merit to making it required, though... I'm not sure.

Reply:
They can always just `new MuxedAccount(gAddress).setId('123');`